source/extensions/filters/network/mysql_proxy/mysql_codec_clogin_resp.cc

```diff
@@ -8,66 +13,145 @@ namespace Extensions {
 namespace NetworkFilters {
 namespace MySQLProxy {
 
-void ClientLoginResponse::setRespCode(uint8_t resp_code) { resp_code_ = resp_code; }
+// https://dev.mysql.com/doc/internals/en/connection-phase-packets.html#packet-Protocol::AuthSwitchRequest
+DecodeStatus AuthSwitchMessage::parseMessage(Buffer::Instance& buffer, uint32_t remain_len) {
+  int origin_len = buffer.length();
+  if (BufferHelper::readUint8(buffer, resp_code_) != DecodeStatus::Success) {
+    ENVOY_LOG(info, "error parsing resp code of auth switch msg");
+    return DecodeStatus::Failure;
+  }
+  ASSERT(resp_code_ == MYSQL_RESP_AUTH_SWITCH);
+  // OldAuthSwitchRequest
+  if (BufferHelper::endOfBuffer(buffer)) {
+    setIsOldAuthSwitch(true);
+    return DecodeStatus::Success;
+  }
+  if (BufferHelper::readString(buffer, auth_plugin_name_) != DecodeStatus::Success) {
+    ENVOY_LOG(info, "error parsing auth plugin name auth switch msg");
+    return DecodeStatus::Failure;
+  }
+  int consumed_len = origin_len - buffer.length();
+  if (BufferHelper::readStringBySize(buffer, remain_len - consumed_len, auth_plugin_data_) !=
+      DecodeStatus::Success) {
+    ENVOY_LOG(info, "error parsing auth plugin data code of auth switch msg");
+    return DecodeStatus::Failure;
+  }
+  setIsOldAuthSwitch(false);
+  return DecodeStatus::Success;
+}
 
-void ClientLoginResponse::setAffectedRows(uint8_t affected_rows) { affected_rows_ = affected_rows; }
+// https://dev.mysql.com/doc/internals/en/packet-OK_Packet.html
+DecodeStatus OkMessage::parseMessage(Buffer::Instance& buffer, uint32_t len) {
+  uint32_t init_len = buffer.length();
+  if (BufferHelper::readUint8(buffer, resp_code_) != DecodeStatus::Success) {
+    ENVOY_LOG(info, "error parsing resp code of ok msg");
+    return DecodeStatus::Failure;
+  }
+  ASSERT(resp_code_ == MYSQL_RESP_OK);
+  if (BufferHelper::readLengthEncodedInteger(buffer, affected_rows_) != DecodeStatus::Success) {
+    ENVOY_LOG(info, "error parsing affected_rows of  ok msg");
+    return DecodeStatus::Failure;
+  }
+  if (BufferHelper::readLengthEncodedInteger(buffer, last_insert_id_) != DecodeStatus::Success) {
+    ENVOY_LOG(info, "error parsing last_insert_id of ok msg");
+    return DecodeStatus::Failure;
+  }
+
+  if (BufferHelper::readUint16(buffer, status_) != DecodeStatus::Success) {
+    ENVOY_LOG(info, "error parsing status of ok msg");
+    return DecodeStatus::Failure;
+  }
+  // the exist of warning field is determined by server cap flag, but a decoder can not know the
+  // cap flag, so just assume the CLIENT_PROTOCOL_41 is always set. ref
+  // https://github.com/mysql/mysql-connector-j/blob/release/8.0/src/main/protocol-impl/java/com/mysql/cj/protocol/a/result/OkPacket.java#L48
+  if (BufferHelper::readUint16(buffer, warnings_) != DecodeStatus::Success) {
+    ENVOY_LOG(info, "error parsing warnings of ok msg");
```

Comment:
Should this be on debug level? I know these were "info", however, don't you think this will be super chatty? The same as all the above error messages I think.

Reply:
Yes, debug level is appropriate.